### PR TITLE
Rajesh

### PR DESCRIPTION
**Pull request by SoloRajesh**
Form Validation and Details send to json-server

### CODE REVIEW

File: src/app/employee-details/add-employee/add-employee.component.css [hunk 1]
```diff
@@ -0,0 +1,90 @@
+body{
+    margin: 20px;
+}
+/* .outer{
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.css]: remove these commented lines

File: src/app/employee-details/add-employee/add-employee.component.css [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+:host >>> .p-dropdown{
+    width: 70%;
+}
+:host >>> .p-calendar{
+    width: 70%;
+}
+:host >>> .p-inputNumber{
+    width: 70%;
+}
+/* :host >>> .p-fileUpload{
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.css]: remove these commented line

File: src/app/employee-details/add-employee/add-employee.component.css [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    :host >>> .p-dropdown{
+        width: 100%;
+    }
+    :host >>> .p-calendar{
+        width: 100%;
+    }
+    :host >>> .p-inputNumber{
+        width: 100%;
+    }
+    /* :host >>> .p-radioButton{
```
Comment (aarthi1246) [inline src/app/employee-details/add-employee/add-employee.component.css]: remove the commended lines

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+  }
+
+  addExperience() {
+    this.yearOfExperience.push(this.experienceDetails());
+  }
+
+  delExperience() {
+    let control = <FormArray>this.myForm.controls['yearOfExperience'];
+    control.removeAt(control.length - 1);
+    // this.yearOfExperience.removeAt(index - 1)
```
Comment (aarthi1246) [inline src/app/employee-details/add-employee/add-employee.component.ts]: remove the commended lines

**Review by aarthi1246 — APPROVED** [review]

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        city: ['', Validators.required],
+        state: ['', Validators.required],
+        pincode: ['', Validators.required],
+      }),
+      fatherName: ['', Validators.required],
+      EmployeeNo: ['', Validators.required],
+      photo: [''],
+      qualification: ['', Validators.required],
+      CTC: ['', Validators.required],
+      email: ['', Validators.required],
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.ts]: set validation for email

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+      experience: ['', Validators.required],
+      yearOfExperience: this.fb.array([
+      ]),
+      emergencyContactNumber: ['', Validators.required],
+      panNumber: ['', Validators.required],
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.ts:56]: PanNumber must be Captialize

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 7]
```diff
@@ -0,0 +1,163 @@
+import { Component } from '@angular/core';
+import {
+  Validators,
+  FormGroup,
+  FormArray,
+  FormBuilder,
+  FormControl,
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.ts]: declared but never used

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+      { department: 'HR' },
+      { department: 'Frontend Developer' },
+      { department: 'Backend Developer' },
+      { department: 'Tester' },
+      { department: 'Team Leader' },
+      { department: 'Cypersecurity' },
+    ];
+  }
+
+  get f() {
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.ts]: give some meaningful name

File: src/app/employee-details/add-employee/add-employee.component.ts [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+    return this.myForm.controls;
+  }
+
+  get address(): FormGroup {
+    return this.myForm.get('address') as FormGroup;
+  }
+
+  get ad() {
```
Comment (HarishDass) [inline src/app/employee-details/add-employee/add-employee.component.ts]: give some meaningful name

**Comment by SoloRajesh** [comment]
i want raise new pr

**Pull request closed without merging**